addOns/oast/src/main/java/org/zaproxy/addon/oast/services/interactsh/InteractshService.java

```diff
@@ -301,18 +313,31 @@ private void schedulePoller(int initialDelay) {
             }
             httpSender.sendAndReceive(pollMsg);
             if (pollMsg.getResponseHeader().getStatusCode() != 200) {
-                LOGGER.warn(
-                        Constant.messages.getString(
-                                "oast.interactsh.error.poll",
-                                Constant.messages.getString("oast.interactsh.error.badHttpCode")));
+                LOGGER.error(
+                        "Error during interactsh poll, due to BadHttpCode {}. Content: {}",
+                        pollMsg.getResponseHeader().getStatusCode(),
+                        pollMsg.getResponseBody().toString());
                 return new ArrayList<>();
             }
+
             JSONObject response = JSONObject.fromObject(pollMsg.getResponseBody().toString());
             if (!response.containsKey("data") || !response.containsKey("aes_key")) {
                 LOGGER.warn(Constant.messages.getString("oast.interactsh.error.poll.badResponse"));
```

Comment:
Unrelated to this PR, shouldn't it return here? Otherwise it will lead to NPE later.